Testnet/MintableTokenInvoice/MintableTokenInvoice/MintableTokenInvoice.cs

```diff
@@ -91,61 +128,114 @@ public string GetInvoiceReference(string transactionReference)
         return $"INV-{invoiceReference.Substring(0, 4)}-{invoiceReference.Substring(4, 4)}-{invoiceReference.Substring(8, 4)}";
     }
 
-    private string ValidateKYC(Address sender, string invoiceReference)
+    private void EnsureKYCdUserOnly(Address sender)
     {
         // KYC check. Call Identity contract.
         ITransferResult result = this.Call(IdentityContract, 0, "GetClaim", new object[] { sender, KYCProvider });
-        if (!(result?.Success ?? false))
-        {
-            string reason = "Could not determine KYC status";
-            Log(new InvoiceResult() { InvoiceReference = invoiceReference, Success = false, Reason = reason });
-            return reason;
-        }
-
-        // The return value is a json string encoding of a Model.Claim object, represented as a byte array using ascii encoding.
-        // The "Key" and "Description" fields of the json-encoded "Claim" object are expected to contain "Identity Approved".
-        if (result.ReturnValue == null || !Serializer.ToString((byte[])result.ReturnValue).Contains("Identity Approved"))
-        {
-            string reason = "Your KYC status is not valid";
-            Log(new InvoiceResult() { InvoiceReference = invoiceReference, Success = false, Reason = reason });
-            return reason;
-        }
+        Assert(result.Success, "Could not determine KYC status");
 
-        return string.Empty;
+        // Only verified users are saved in the Identity contract.
+        Assert(result.ReturnValue != null && ((byte[])result.ReturnValue).Length != 0, "Your KYC status is not valid");
     }
 
-    /// <inheritdoc />
-    public string CreateInvoice(string symbol, UInt256 amount, UInt128 uniqueNumber)
+    private string CreateInvoiceInternal(Address address, string symbol, UInt256 amount, UInt256 fee, UInt128 uniqueNumber, string targetAddress, string targetNetwork)
```

Comment:
There are two methods passing basically the same set of arguments. In one case we pass the parameters inside of a url and in the other case they are passed as separate values. In both cases, for consistency, we should pass the same values. Since we can't strip the "fee" from the url we will be passing it here too even if it does not make 100% sense. Its simply for consistency to keep the flows in the two methods as similar as possible. See it as a check to ensure that the fee the user agreed on is still the same as what is configured on the contract. On extreme cases it could be different and the call should fail.